### DISCUSSION TITLE
database restore on mariadb throws ERROR at line 1: ASCII '\0' appeared in the statement

### DISCUSSION THREAD

**Discussion by usma0118** [tiredofit/docker-db-backup#360] [Q&A]
I have tried restoring a backup taken from mariadb server, and tried restoring it into new db on same server. it's throwing error:

`ERROR at line 1: ASCII '\0' appeared in the statement, but this is not allowed unless option --binary-mode is enabled and mysql is run in non-interactive mode. Set --binary-mode to 1 if ASCII '\0' is expected.`

File is encrypted and compressed.

filename: mariadb_mydb_mariadb.default.svc.cluster.local_20240811-132308.sql.zst.gpg

Any ideas?

**Comment by tiredofit** [comment]
Hmm - I can't quite remember if I built in decryption routines into the restore script. You would do best to decrypt it manually. I'll take a peek tomorrow to refresh my memory on this..

**Reply by usma0118** [reply]
Maybe I am naive… I should be able to use
`gpg --output filename --decrypt file.gpg` to decrypt?

Since I am using passphrase only (symmetric)?

I can see encryption code snippet here: https://github.com/tiredofit/docker-db-backup/blob/90f53a7f00d7e26e6622c53e594b8de103735eef/install/assets/functions/10-db-backup#L1312

at moment, i am getting gpg: decryption failed: Bad session key

**Reply by tiredofit** [reply]
Interesting. `gpg --output <output> --decrypt <your backup>` should give you a passphrase prompt. The snippet you posted above is the correct encryption routine..

**Reply by usma0118** [reply]
Tested on two different containers with same encryption key.

Without encryption: back is created and restorable. 

With gpg encryption and passphrase: unable to decrypt.

**Reply by usma0118** [reply]
from container, i have used your snippet and encrypted/decrypted a test file successfully with said encryption key

**Reply by usma0118** [reply]
Root cause: https://github.com/tiredofit/docker-db-backup/issues/362
